src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleOpaqueTokenResolver.scala

```diff
@@ -55,15 +61,17 @@ class StandardGoogleOpaqueTokenResolver(directoryDAO: DirectoryDAO, googleTokenI
       }
     }
 
-  private def lookupUser(maybeTokenInfoResponse: Option[TokenInfoResponse], samRequestContext: SamRequestContext): IO[Option[WorkbenchUser]] = {
+  private def lookupUser(maybeTokenInfoResponse: Option[TokenInfoResponse], samRequestContext: SamRequestContext): IO[Option[GoogleTokenInfo]] = {
     maybeTokenInfoResponse.map {
-      case TokenInfoResponse(googleSubjectId, email) =>
+      case TokenInfoResponse(googleSubjectId) =>
         directoryDAO.loadSubjectFromGoogleSubjectId(googleSubjectId, samRequestContext).map(_.flatMap {
-          case userId: WorkbenchUserId => Option(WorkbenchUser(userId, Option(googleSubjectId), email, None))
-          case _ => None
+          case userId: WorkbenchUserId => Option(GoogleTokenInfo(Option(userId), googleSubjectId))
+          case PetServiceAccountId(userId, _) => Option(GoogleTokenInfo(Option(userId), googleSubjectId))
```

Comment:
Should we block pets here?

Reply:
I am happy for other opinions here. At the moment I don't think pets can actually make it here because they can't authenticate via azure b2c. If that were to change, would this code be right? I think my mind is changing here.

Reply:
I removed this line